packages/esm-active-visits-app/src/active-visits-widget/active-visits.resource.tsx

```diff
@@ -37,18 +39,89 @@ export function useActiveVisits() {
     openmrsFetch,
   );
 
-  const mapVisitProperties = (visit: Visit): ActiveVisit => ({
-    age: visit?.patient?.person?.age,
-    id: visit.uuid,
-    idNumber: visit?.patient?.identifiers[0]?.identifier,
-    gender: visit?.patient?.person?.gender,
-    location: visit?.location?.uuid,
-    name: visit?.patient?.person?.display,
-    patientUuid: visit?.patient?.uuid,
-    visitStartTime: visit?.startDatetime,
-    visitType: visit?.visitType?.display,
-    visitUuid: visit.uuid,
-  });
+  const mapVisitProperties = (visit: Visit): ActiveVisit => {
+    //create base object
+    let activeVisits: ActiveVisit = {
+      age: visit?.patient?.person?.age,
+      id: visit.uuid,
+      gender: visit?.patient?.person?.gender,
+      location: visit?.location?.uuid,
+      name: visit?.patient?.person?.display,
+      patientUuid: visit?.patient?.uuid,
+      visitStartTime: visit?.startDatetime,
+      visitType: visit?.visitType?.display,
+      visitUuid: visit.uuid,
+    };
+
+    //in case no configuration is given the previsous behavior remanes the same
+    if (!config?.activeVisits?.identifiers) {
+      config.activeVisits.identifiers = [
+        {
+          header: {
+            key: 'idNumber',
+            default: 'ID Number',
+          },
+          identifierName: visit?.patient?.identifiers[0].identifierType?.name,
+        },
+      ];
+    }
+
+    //map identifires on config
+    config?.activeVisits?.identifiers?.map((configIdentifier) => {
+      //check if in the current visit the patient has in his identifiers the current identifierType name
+      const visitIdentifier = visit?.patient?.identifiers.find((visitIdentifier) => {
+        if (visitIdentifier?.identifierType?.name === configIdentifier?.identifierName) {
+          return true;
+        }
+        return false;
+      });
+
+      if (visitIdentifier) {
+        //add the new identifier or rewrite existing one to activeVisit object
+        //the parameter will corresponde to the name of the key value of the configuration
+        //and the respective value is the visit identifier
+
+        activeVisits = {
+          ...activeVisits,
+          [configIdentifier?.header?.key]: visitIdentifier?.identifier,
+        };
```

Comment:
```suggestion
        activeVisits[configIdentifier.header?.key]: visitIdentifier.identifier;
```

Reply:
Heck, this whole `if` could be:

```suggestion
activeVisits[configIdentifier.header?.key]: visitIdentifier?.identifier ?? '--';
```